docs/docs/glossary.md

```diff
@@ -2,7 +2,272 @@
-pull request gets accepted.
+## A
+
+### API
+
+Application Programming Interface: A method for one application to communicate with another. For example, a [source plugin](#source-plugin) will often use an API to get its data.
+
+### Accessibility
+
+The inclusive practice of removing barriers that prevent interaction with, or access to websites, by people with disabilities. When sites are correctly designed, developed and edited for accessibility, generally all users have equal access to information and functionality. Read about [Gatsby's Commitment to Accessibility](/blog/2019-04-18-gatsby-commitment-to-accessibility/).
+
+## B
+
+### Babel
+
+A tool that lets you write the most modern [JavaScript](#javascript), and on [build](#build) it gets [compiled](#compiler) to code that most web browsers can understand.
+
+### Back-End
+
+The behind the scenes that the [public](#public) do not see. This often refers to the control panel of your [CMS](#cms). These are often powered by server-side programming languages such as PHP or ASP.net.
+
+### Build
+
+The process of taking your code and content and packaging it into a website that can be hosted and accessed.
+
+## C
+
+### CLI
+
+Command Line Interface: An application that runs on your computer through the [command line](#command-line) and interacted with your keyboard.
+
+Gatsby has two command line interfaces. One, [`gatsby`](/docs/gatsby-cli/), for day-to-day development with Gatsby and another, [`gatsby-dev`](/contributing/setting-up-your-local-dev-environment/#gatsby-repo-install-instructions), for those who contribute to the Gatsby project.
+
+### CMS
+
+Content Management System: an application where you can manage your content and have it saved to a database or file for accessing later. Examples of Content Management Systems include Wordpress, Drupal, Contentful, and Netlify CMS.
+
+### Command Line
+
+A text-based interface to run commands on your computer. The default Command Line applications for Mac and Windows are `Terminal` and `Command Prompt` respectively.
+
+### Compiler
+
+A compiler is a program that translates code written in one language to another language. For example [Gatsby](#gatsby) can compile [React](#react) applications into static [HTML](#html) files.
+
+### Components
+
+Components are independent and re-usable chunks of code powered by [React](#react) that, when combined, make up your website or app.
+
+A component can include components within it. In fact, [pages](#page) and [templates](#templates) are examples of components.
+
+### Config
+
+The configuration file, `gatsby-config.js` tells Gatsby information about your website. A common option set in config is your sites meta data that can power your SEO meta tags.
+
+## D
+
+### Data Source
+
+A source of data that usually gets fed into Gatsby using [source plugins](#source-plugin). A data source is often a [Headless CMS](#headless-cms), but it could also include Markdown, JSON, or YAML files.
```

Comment:
```suggestion
Content and data's origin point, typically integrated into Gatsby with [source plugins](#source-plugin). A data source is often a [Headless CMS](#headless-cms), but it could also include Markdown, JSON, or YAML files.
```